### PR TITLE
fixes number overflow

### PR DESCRIPTION
**Pull request by brendansudol**
fixes: https://github.com/18F/cms-hitech-apd/issues/935

preview:
screen shot 2018-08-14 at 3 18 02 pm

### CODE REVIEW

File: web/src/containers/activity/CostAllocateFFP.js [hunk 1]
```diff
@@ -37,23 +37,25 @@ class CostAllocateFFP extends Component {
...
                   <DollarInput
+                    wrapperClass="lg-col-8 mb2"
                     name={`cost-allocate-other-${year}`}
                     label={t('activities.costAllocate.ffp.labels.other')}
                     value={costAllocation[year].other}
                     onChange={this.handleOther(year)}
                   />
                   <div>{t('activities.costAllocate.ffp.totalNetOther')}</div>
-                  <div className="h3 bold mono">
+                  <div className="h4 bold mono truncate">
```
Comment (mgwalker) [inline web/src/containers/activity/CostAllocateFFP.js]: Question for @lauraponce @quarterback @NAretakis @jeromeleecms:

Is truncating these dollar numbers okay if they get too big?  (It looks like `$123,345,679...`)  It looks like the cost allocation component can handle into the hundreds of millions before it truncates so ***hopefully*** that should be fine.  (Goodness I hope you don't see many hundred million dollar APDs...)

Comment (brendansudol) [reply web/src/containers/activity/CostAllocateFFP.js]: i think the solution for really big numbers just not be truncating but utilizing SI prefix (https://en.wikipedia.org/wiki/Metric_prefix#List_of_SI_prefixes) -- for example, `$123.45B` or `1.21M`

**Comment by brendansudol** [comment]
@mgwalker i'll fix those budget tables ones here too, thanks for heads up about that

**Comment by brendansudol** [comment]
ok, i adjusted this to better accommodate big money numbers (right now that threshold is a million dollars so it's very easy to adjust).

if the number is less than a cool mil, it gets the standard treatment: `$999,999`
if the number is better, we utilize [SI prefixes](https://en.wikipedia.org/wiki/Metric_prefix#List_of_SI_prefixes) with at most 3 significant digits: `$1.23M` (if there are trailing decimal 0s, they get removed, as in `$8.1M` vs `$8.10M`)

preview:
image

**Comment by mgwalker** [comment]
I like the postfixed dollar values a lot.  Definitely want product (cc: @jeromeleecms) to weigh in, though.  To be clear, this only changes how the dollar values are displayed, not what we're actually storing, so the analyst/reviewer view could still show full dollar values.

**Comment by quarterback** [comment]
Good solution

**Pull request merged**